### ISSUE TITLE
Fix hardcoded CUDA versions to folder name autodetection?

### ISSUE THREAD

**Issue by sairus7** [JuliaGPU/CUDAapi.jl#66]
On my machine could not identify CUDA libraries because of new CUDA toolkit version 10.1 absent in this list:
```
const cuda_versions = Dict(
    "toolkit"   => [v"1.0", v"1.1",
                    v"2.0", v"2.1", v"2.2",
                    v"3.0", v"3.1", v"3.2",
                    v"4.0", v"4.1", v"4.2",
                    v"5.0", v"5.5",
                    v"6.0", v"6.5",
                    v"7.0", v"7.5",
                    v"8.0",
                    v"9.0", v"9.1", v"9.2",
                    v"10.0"],
    "cudnn"     => [v"1.0",
                    v"2.0",
                    v"3.0",
                    v"4.0",
                    v"5.0", v"5.1",
                    v"6.0",
                    v"7.0", v"7.1", v"7.3", v"7.4"]
)
```
Maybe it will be simpler to autodetect library versions based on, e.g. folder name or from direct library call, instead of adding every new version to this list manually?

**Comment by maleadt** [comment]
This isn't as easy as it sounds. Instead of doing `Libdl.find_library` on a couple of library names (i.e. appending possible version strings) we'd need to redo all that logic and actively scan the system library paths for candidate libraries and extract a version tag.

Alternatively, we could have some functionality that scans the system and returns candidate version numbers for use with the existing infrastructure in CUDAapi, but then we'd be duplicating functionality (that hard codes, e.g., the paths where CUDA is installed, or where `libcudnn` could reside).

In summary, I think it's much easier to just maintain a version list and bump them when necessary.
If you think otherwise, please go ahead and propose a change :slightly_smiling_face:

**Comment by maleadt** [comment]
Besides, there's other compatibility information to be extracted from a new CUDA release which is even harder to automate: https://github.com/JuliaGPU/CUDAapi.jl/pull/67
